### PR TITLE
docs: Improvements to HPC launcher docs

### PR DESCRIPTION
**Pull request by jerryharrow**
## Description

* Provide inline info about agent-specific scheduling options that do not apply to HPC Launcher configurations.
* Identify enroot-specific differences from docker (like for Singularity)
* Provide reference to custom resource pools as an option to deal with non-homogenous Slurm/PBS partitions.

## Test Plan

==

Passed all wording changes through grammerly.com

image

==

image

==

image

==

image

==

image

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst [hunk 1]
```diff
@@ -238,6 +243,20 @@ sometimes resolved by additionally installing the ``apptainer-setuid`` package.
 -  Enroot does not provide a mechanism for sharing containers. Each user must create any containers
    needed by their Determined experiments prior to creating the experiment.
 
+-  Some Docker features do not have an exact replacement in Enroot.
```
Comment (rcorujo) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst:246]: The features listed below are Docker features?  Aren't they Determined features, not Docker?

Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-known-issues.rst:246]: They are Determined features that support the use of Docker features that are not in Enroot.

This is the same wording we used above for Singularity.

I mean we could say `Some Docker features do not have an exact replacement in Enroot and therefore the associated Determined features are not supported`?

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst [hunk 2]
```diff
@@ -188,6 +195,13 @@ to optimize how Determined interacts with PBS:
       using the :ref:`pbs section <cluster-configuration-slurm>` ``default_compute_resource_pool``
       or ``default_aux_resource_pool`` option.
 
+   -  If a PBS queue is not homogeneous, you may create a resource pool that provides homogenous
+      resources out of that queue using a custom resource pool. Configure a :ref:`resource pool
+      <cluster-resource-pools>` with ``provider_type: hpc``, specify the underlying PBS queue name
+      to receive the job and include a :ref:`task_container_defaults
+      <master-task-container-defaults>` section with the necessary ``psb`` options to select the
```
Comment (jagadeesh545) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:202]: knit - typo. should be "pbs"

**Review by rcorujo — APPROVED** [review]

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst (hunk 2 above)
Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:202]: Good catch.  Grammerly doesn't help much with that :-)

**Pull request merged**